lib/saluki-components/src/destinations/datadog/common/io.rs

```diff
@@ -254,7 +254,22 @@ async fn run_endpoint_io_loop<S, B>(
         ))
         .service(service.map_err(|e| e.into()));
 
-    let retry_queue = RetryQueue::new(endpoint_url.clone(), config.retry().queue_max_size_bytes());
+    let mut retry_queue = RetryQueue::new(endpoint_url.clone(), config.retry().queue_max_size_bytes());
+
+    // If the storage size is set, enable disk persistence for the retry queue.
+    if config.retry().storage_max_size_bytes() > 0 {
+        retry_queue = retry_queue
+            .with_disk_persistence(
+                PathBuf::from(config.retry().storage_path()),
+                config.retry().storage_max_size_bytes(),
+                config.retry().storage_max_disk_ratio(),
+                Arc::new(DiskUsageRetrieverImpl::new(PathBuf::from(
+                    config.retry().storage_path(),
+                ))),
+            )
+            .await
+            .expect("Failed to initialize disk persistence for retry queue.");
```

Comment:
Hmmm, the not so great part is that we would just be panicking and killing the endpoint I/O task here.

What does the Agent do in terms of error handling? Is there error handling during the _construction_ of the on-disk retry queue?

Reply:
For context, what I'm trying to suss out is if we need to be pushing this error up the stack to end up killing the process in the same way we would when encountering any other sort of start-up error.

Reply:
Looks like it continues but just without disk persistence. https://github.com/DataDog/datadog-agent/blob/main/comp/forwarder/defaultforwarder/internal/retry/transaction_retry_queue.go#L61-L67

Updated the code to remove the expect

Reply:
Alright, neat, that makes things simpler. 👍🏻